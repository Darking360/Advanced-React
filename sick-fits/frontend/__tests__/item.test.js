import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const FakeItem = {
  id: 'ABC123',
  title: 'Cool item',
  description: 'Cool description',
  price: 10000,
  image: 'dog.jpg',
  largeImage: 'dog_large.jpg', 
};

describe('<Item />', () => {

  it('render and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={FakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();

  });
  it('renders the price tag and title', () => {
    const wrapper = shallow(<ItemComponent item={FakeItem} />);
    const PriceTag = wrapper.find('PriceTag');
    expect(PriceTag.children().text()).toBe('$100');
    expect(wrapper.find('Title a').text()).toBe('Cool item');
    
  });

  it('renders image propertly', () => {
    const wrapper = shallow(<ItemComponent item={FakeItem} />);
    const Image = wrapper.find('img');
    expect(Image.props().src).toBe('dog.jpg');
    expect(Image.props().alt).toBe('Cool item');
  });

  it('renders the buttons propertly', () => {
    const wrapper = shallow(<ItemComponent item={FakeItem} />);
    const buttons = wrapper.find('.buttonList');
    expect(buttons.children()).toHaveLength(3);
    expect(buttons.find('Link').exists()).toBe(true);
    expect(buttons.find('Link')).toHaveLength(1);// Another way to do it
    expect(buttons.find('AddToCart').exists()).toBe(true);
    expect(buttons.find('DeleteItem').exists()).toBe(true);
  });
});