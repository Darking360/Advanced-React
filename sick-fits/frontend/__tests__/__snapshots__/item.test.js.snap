// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> render and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Cool item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $100
  </PriceTag>
  <p>
    Cool description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    />
  </div>
</ItemStyles__Item>
`;
